pulsar-broker/src/main/java/com/yahoo/pulsar/broker/PulsarService.java

```diff
@@ -565,41 +567,40 @@ public MessagingServiceShutdownHook getShutdownService() {
     /**
      * Derive the host
      *
-     * @param isBindOnLocalhost
-     * @return
+     * @return Hostname or IP address the service binds on.
      */
     public static String host(ServiceConfiguration config) {
-        try {
-            if (!config.isBindOnLocalhost()) {
-                return InetAddress.getLocalHost().getHostName();
-            } else {
-                return "localhost";
-            }
-        } catch (Exception e) {
-            LOG.error(e.getMessage(), e);
-            throw new IllegalStateException("failed to find host", e);
-        }
+        return config.getBindAddress();
```

Comment:
These 2 methods should have the above logic to check whether bindAddress and advertisedAddress were configured.